src/main/java/org/variantsync/functjonal/list/FilterMappedList.java

```diff
@@ -0,0 +1,229 @@
+package org.variantsync.diffdetective.util;
+
+import java.util.AbstractSequentialList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+/**
+ * Unmodifiable view of a list where the elements are filtered using a predicate and transformed
+ * using a mapping function.
+ *
+ * <p>Modifications to the backing list are directly visible to this list, although all iterators are
+ * invalid as soon a modification occurs (regardless of what additional promises the underlying
+ * list implementation makes).
+ *
+ * <p>This view only provides sequential access to the underlying list (to make changes to the
+ * backing list visible immediately). This implies that all random access methods and {@link size}
+ * run in linear time, therefore specialized functions like {@code isEmpty} and iteration over the
+ * list should be preferred.
+ *
+ * <p>The filter and map function provided to this class may be called at arbitrary times and for an
+ * arbitrary number of times for each element in the backing list. To get predictable behaviour and
+ * allow future modifications to this implementation, the filter and map function should therefore
+ * be a pure function: It should have no side effects and return the same result given the same
+ * element. What <i>the same</i> means (identity, equality or something different) is up to the user
+ * of this class, as this implementation doesn't make any assumption about this.
+ *
+ * @author Benjamin Moosherr
+ */
+public class FilterMappedList<Orig, New> extends AbstractSequentialList<New> {
```

Comment:
As this is a view and not itself a list, name it as such? What about `FilteredMappedListView`?

Reply:
Yes, that's probably a good idea. Although we should keep the length of our names in mind, as this is now very long.